docs/developer_documentation/api-guidelines.md

```diff
@@ -0,0 +1,49 @@
+# API Design Guidelines
+
+When writing an API for a new module or feature, strive to keep it small and concise. 
+
+We use Swagger to document our APIs. See the [Swagger reference](ref-swagger.md) for more information on how to set this 
+up for your module.
+
+### Public vs internal APIs
+There's a difference between public and internal APIs: internal APIs are tailor made for specific plugins and are subject
+to change. Public APIs are stable and versioned and can be used by third parties when integrating with MOLGENIS.
+
+### API naming convention
+When adding a new API, we adhere to the following naming scheme:
+
+Public APIs:
+`/api/<api_human_readable_identifier>/v<api_version>/*`
+
+Internal APIs:
+`/api/internal/<api_human_readable_identifier>/*`
+Internal APIs don't have a version because the corresponding plugins will evolve together with the API.
+
+For example:
+
+| API | URI | Endpoints |
+|:---|:---|:---------|
+| Authentication API | `/api/auth/v1` | `/login`, `/logout` |                    
+| REST API | `/api/rest/v1` | REST CRUD operations on resources and resource collections |
+| Metadata API | `/api/internal/meta/v1` | ... |
```

Comment:
unless the api/rest/ will expose metadata via an endpoint, I would like to have this as a public API

Reply:
Would you agree that the current metadata editing api is still an internal api?

Reply:
Yes that is true, but if this is a list of "possible how we want it to be" endpoints we should keep this in mind

Reply:
@ConnorStroomberg interesting! let's talk!